ui/components/multichain/pages/gator-permissions/review-permissions/review-gator-permissions-page.tsx

```diff
@@ -89,39 +126,65 @@ export const ReviewGatorPermissionsPage = ({
     chainId: (chainId ?? '') as Hex,
   });
 
-  useEffect(() => {
-    setTotalGatorPermissions(gatorPermissions.length);
-  }, [chainId, gatorPermissions]);
+  const handleRevokeClick = useCallback(
+    async (
+      permission: StoredGatorPermissionSanitized<
+        Signer,
+        PermissionTypesWithCustom
+      >,
+    ) => {
+      const { context } = permission.permissionResponse;
 
-  const handleRevokeClick = async (
-    permission: StoredGatorPermissionSanitized<
-      Signer,
-      PermissionTypesWithCustom
-    >,
-  ) => {
-    try {
-      await revokeGatorPermission(permission);
-    } catch (error) {
-      console.error('Error revoking gator permission:', error);
-    }
-  };
+      // Set pending state immediately to disable button and show "Pending..." text
+      addPendingContext(context);
+
+      try {
+        await revokeGatorPermission(permission);
+
+        // Clear any existing timeout for this context
+        const existingTimeout = revokeTimeoutsRef.current.get(context);
```

Comment:
Feel free to answer this and otherwise ignore: I don't fully understand why this is necessary. Once the transaction window shows, wouldn't this view be unmounted, at which point the state would be discarded? Do we need to explicitly remove the "pending revocation click"?